integration/helpers.go

```diff
@@ -348,3 +352,21 @@ func RunAPIServerProxy(closeCh <-chan struct{}) error {
 	}()
 	return nil
 }
+
+// IsFreshStatusConditionTrue returns true when conditionType is present, set to `metav1.ConditionTrue`
+// and `.ObservedGeneration` matches `addon.ObjectMeta.Generation`
+func IsFreshStatusConditionTrue(addon *addonsv1alpha1.Addon, conditionType string) bool {
```

Comment:
Can we wrap this in a `WaitForObjectCondition(addon, conditionType, Status)`?
In the two places we are using this code, we are using it within a wait anyway.

Please also drop the `Fresh` from the function name, Conditions should always be checked against `.ObservedGeneration`, I don't want another function called `IsStaleStatusConditionTrue`? :)

Reply:
I named it that way to distinguish it from apimachinery's builtin `IsStatusConditionTrue` [1] which does not compare `.ObservedGeneration`.

[1] https://github.com/kubernetes/apimachinery/blob/v0.22.4/pkg/api/meta/conditions.go#L85

Reply:
done!